pnet_packet/Cargo.toml

```diff
@@ -0,0 +1,29 @@
+[package]
+name = "pnet_packet"
+version = "0.19.0"
+authors = ["Robert Clipsham <robert@octarineparrot.com>"]
+license = "MIT/Apache-2.0"
+homepage = "https://github.com/libpnet/libpnet"
+repository = "https://github.com/libpnet/libpnet"
+description = "Cross-platform, binary packet parsing and manipulation"
+keywords = ["networking", "packet", "parsing", "raw"]
+categories = ["network-programming", "parser-implementations"]
+build = "build.rs"
+
+[features]
+default = ["with-syntex"]
+nightly = ["pnet_macros_plugin"]
+with-syntex = ["syntex", "pnet_macros/with-syntex"]
+
+[dependencies]
+pnet_base = { path = "../pnet_base" }
+pnet_macros = { path = "../pnet_macros", optional = true, version = ">=0.9" }
+pnet_macros_support = { path = "../pnet_macros_support", version = ">=0.1" }
+pnet_macros_plugin = { path = "../pnet_macros_plugin", optional = true, version = ">=0.1" }
```

Comment:
I have not bothered to learn the difference between the three `pnet_macros.*` crates. How come this "top level" crate must depend on all of them? If they are support crates to `pnet_macros` would they not be pulled in by `pnet_macros` for us?

Reply:
* `pnet_macros` is the important one with all the logic for expanding `#[packet]`
 * `pnet_macros_plugin` has the `plugin = true` flag set, and registers a compiler plugin with a nightly rustc
 * `pnet_macros_support` provides type aliases for `u16le` etc. Any code generated by `pnet_macros` requires it.

`pnet_packet` requires either `pnet_macros` + syntex, or `pnet_macros_plugin` (which hasn't compiled in a long time because we target a rather old version of syntex... And I'm not sure syntex is even kept up to date with rustc now, but that's another issue...)

I think I should be able to get rid of the plugin/macros dependencies - they should only be build time dependencies. I'll check and see.

Reply:
Awesome if this could be simplified indeed! Also, it sounds like `pnet_macros_support` should be a dependency of `pnet_macros` so we don't need it here?